src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,212 @@
+import me.zeroeightsix.kami.mixin.extend.itemUseCooldown
+import me.zeroeightsix.kami.setting.ImGuiExtra
+import me.zeroeightsix.kami.util.InstrumentMap
+import me.zeroeightsix.kami.util.MidiParser
+import me.zeroeightsix.kami.util.Note
+import me.zeroeightsix.kami.util.Viewblock.getIrreplaceableNeighbour
+import me.zeroeightsix.kami.util.plus
+import me.zeroeightsix.kami.util.text
+import net.minecraft.block.Blocks
+import net.minecraft.block.NoteBlock.INSTRUMENT
+import net.minecraft.block.NoteBlock.NOTE
+import net.minecraft.block.enums.Instrument
+import net.minecraft.client.network.ClientPlayerEntity
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket
+import net.minecraft.util.Hand
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Vec3d
+import net.minecraft.world.World
+import java.io.File
+import java.util.ArrayList
+import java.util.TreeMap
+
+@Module.Info(
+    name = "Notebot",
+    category = Module.Category.MISC,
+    description = "Plays music with noteblocks; put songs as .mid files in .minecraft/songs"
+)
+object Notebot : Module() {
+
+    private var lastNote: Long = 0L
+    private var elapsed: Long = 0L
+    var playingSong = false
+
+    private var noteSequence: TreeMap<Long, ArrayList<Note>> = TreeMap()
+    var instrumentMap = InstrumentMap()
+
+    var channelMap: TreeMap<Int, Instrument> = TreeMap()
+
+    /* This is a bad way to have channels. In the future, I would like for there to be some way to display a map.*/
+    @Setting(name = "Channel 0")
+    var channelZero = Instrument.HAT
+
+    @Setting(name = "Channel 1")
+    var channelOne = Instrument.HAT
+
+    @Setting(name = "Channel 2")
+    var channelTwo = Instrument.HAT
+
+    @Setting(name = "Channel 3")
+    var channelThree = Instrument.HAT
+
+    @Setting(name = "Channel 4")
+    var channelFour = Instrument.HAT
+
+    @Setting
+    var hatAlwaysAsFSharp = true
+
+    @Setting(comment = "Name of MIDI file in the songs folder of your .minecraft directory")
+    @ImGuiExtra.Post("renderButtons")
+    var songName = "tetris_theme.mid"
+
+    fun renderButtons() {
+        dsl.button("Discover") {
+            discoverSong()
+        }
+        dsl.button("Load") {
+            loadSong()
+        }
+
+        ImGui.pushItemFlag(ItemFlag.Disabled.i, true)
+        ImGui.pushStyleVar(StyleVar.Alpha, ImGui.style.alpha * 0.5f)
+
+        ImGui.popItemFlag()
+        ImGui.popStyleVar()
+        playButton()
+        pauseButton()
+    }
+
+    private fun playButton() {
+        (playingSong).conditionalWrap(
+            {
+                ImGui.pushItemFlag(ItemFlag.Disabled.i, true)
+                ImGui.pushStyleVar(StyleVar.Alpha, ImGui.style.alpha * 0.5f)
+            },
+            {
+                dsl.button("Play") {
+                    playingSong = true
+                }
+            },
+            {
+                ImGui.popItemFlag()
+                ImGui.popStyleVar()
+            }
+        )
+    }
```

Comment:
Because this same conditionalWrap occurs three times in the codebase now (2 here, 1 in ESP), make an util method in `Utils` to deduplicate this code. You probably want the call to look something like
```kotlin
wrapDisabled(playingSong) {
    dsl.button("Play") {
        playingSong = true
    }
}
```
Replace those three occurences with your util method.